Bottom: d1daff64a4f75738e31ae3157a0d0b2a258a4613
Top:    f747cb4a6d65cca424ec8acc08af1bd3c69110d3
Author: Parth Shah <parth@linux.ibm.com>
Date:   2020-05-14 09:47:06 +0530

Refresh of sched-allow-sched_-get-set

---

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index 44bcbf060718..5b89f25ad924 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -4744,14 +4744,13 @@ static void __setscheduler(struct rq *rq, struct task_struct *p,
 	if (attr->sched_flags & SCHED_FLAG_KEEP_PARAMS)
 		return;
 
-	__setscheduler_params(p, attr);
-
 	/*
-	 * Change latency_nice value only when SCHED_FLAG_LATENCY_NICE or
-	 * SCHED_FLAG_ALL sched_flag is set.
+	 * Allow changing latency_nice value independently of static_prio
 	 */
 	if (attr->sched_flags & SCHED_FLAG_LATENCY_NICE)
 		p->latency_nice = attr->sched_latency_nice;
+	else if (attr->sched_flags != SCHED_FLAG_LATENCY_NICE)
+		__setscheduler_params(p, attr);
 
 	/*
 	 * Keep a potential priority boosting if called from
